xsd-parser-cli/src/main.rs

```diff
@@ -58,32 +60,33 @@ fn process_dir(input_path: &Path, output_path: &Path) -> Result<(), String> {
     Ok(())
 }
 
-fn process_single_file(input_path: &Path, output_path: Option<&str>) -> Result<(), String> {
+fn process_single_file(input_path: &Path, output_path: Option<&str>) -> anyhow::Result<()> {
     let text = load_file(input_path)?;
-    let rs_file = parse(text.as_str()).map_err(|_| "Error parsing file".to_string())?;
+    let rs_file = parse(text.as_str()).unwrap();
     let gen = GeneratorBuilder::default().build();
     let code = gen.generate_rs_file(&rs_file);
     if let Some(output_filename) = output_path {
-        write_to_file(output_filename, &code).map_err(|e| format!("Error writing file: {}", e))?;
+        write_to_file(output_filename, &code).context("Error writing file")?;
     } else {
         println!("{}", code);
     }
     Ok(())
 }
 
-fn load_file(path: &Path) -> Result<String, String> {
-    let mut file = fs::File::open(&path).map_err(|e| e.to_string())?;
+fn load_file(path: &Path) -> anyhow::Result<String> {
+    let mut file = fs::File::open(&path)?;
     let mut text = String::new();
-    file.read_to_string(&mut text).map_err(|e| e.to_string())?;
+    file.read_to_string(&mut text)?;
     Ok(text)
 }
 
-fn write_to_file(path: &str, text: &str) -> Result<(), String> {
+fn write_to_file(path: &str, text: &str) -> anyhow::Result<()> {
```

Comment:
Looks to me like this function can just return `io::Result`; same above. Or maybe we should use `fs_err` and its error type instead for better error messages.